src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp

```diff
@@ -258,6 +258,45 @@ StatefulSDPAFusion::StatefulSDPAFusion() {
         else
             assign_v_node->set_arguments({new_node->output(2)});
 
+        // Markup pattern:
+        // ReadValue->Convert(Optional)->ScaledDotProductAttentionWithKVCache->Convert(Optional)->Assigin, so that
+        // ReadValue can't be replaced with ReadValueWithSubgraph in this pattern.
+        // TODO: Temporarily skip this pattern. If MemoryInputSDPA supports Subgraph in the future, it may be deleted.
+        for (auto inp : args) {
+            auto rv = as_type_ptr<ov::opset6::ReadValue>(inp.get_node_shared_ptr());
+
+            // Check if it is Convert
+            if (!rv) {
+                auto convert = as_type_ptr<ov::op::v0::Convert>(inp.get_node_shared_ptr());
+                if (convert && convert->get_input_size() > 0u) {
+                    rv = as_type_ptr<ov::opset6::ReadValue>(convert->get_input_source_output(0).get_node_shared_ptr());
+                }
+            }
+
+            if (rv) {
+                for (size_t port = 0; port < new_node->get_output_size(); port++) {
+                    for (const auto& child : new_node->get_output_target_inputs(port)) {
+                        auto assign = ov::as_type_ptr<ov::op::v6::Assign>(child.get_node()->shared_from_this());
+                        if (!assign) {
+                            auto convert = ov::as_type_ptr<ov::op::v0::Convert>(child.get_node()->shared_from_this());
+                            if (convert && convert->get_output_size() > 0) {
+                                assign = ov::as_type_ptr<ov::op::v6::Assign>(
+                                    convert->get_output_target_inputs(0).begin()->get_node()->shared_from_this());
+                            }
+                        }
+
+                        if (assign) {
+                            if (assign->get_variable_id() == rv->get_variable_id()) {
+                                // Markup to runtime info
+                                rv->get_rt_info()["DisableInitSubgraphFusing"] = true;
```

Comment:
Maybe we can try using `past_k_node`/`past_v_node` directly.

Reply:
As you said, `past_k_node`, `past_v_node` should remain unchanged before/after replacing node. Updated.